courses/machine_learning/deepdive/08_image/mnist_linear.ipynb

```diff
@@ -9,9 +9,7 @@
    "source": [
```

Comment:
<ul><li>For readability &amp; simplicity, I'd import the layers and model (Flatten, Dense, Sequential etc.) at the beginning of the notebook, allowing us to have super code to describe the model. I'd also add to set the "input_shape" parameter to the Flatten layer so that we can then do a "model.summary()" from what this function returns. This would look something like:</li></ul><p> </p><p>&nbsp;&nbsp;model = Sequential([</p><p>&nbsp;&nbsp;&nbsp;&nbsp;Flatten(input_shape=[HEIGHT, WIDTH]),</p><p>&nbsp;&nbsp;&nbsp;Dense(NCLASSES),</p><p>&nbsp;&nbsp;&nbsp;&nbsp;Softmax()</p><p>&nbsp;&nbsp;])</p><p> </p><ul><li>I am not understanding the comment</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/551/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>Ah, good idea. The next notebook in the series expands on this, so to keep the experience consistent, I shortened it to layers.&lt;layer-name&gt;. There's lots of different layers in the next one, so there would be a lot of imports. How's that for a compromise?</p><p> </p><p>As for the comment, good call I don't understand it either. Axed!</p>

Reply:
<p>Personally, I prefer to have more complex imports and shorter/simpler code because you spend time understanding the code, not the imports ( and also the layer code can become quite long with all the parameters). Also, having imports like the following is not too ugly and gives at the very beginning of the notebook all the layers we are gonna use in the notebook as a nice compact menu:</p><p> </p><p>from tensorflow.keras.layers import (</p><p>  Dense,</p><p>  Flattern,</p><p> Softmax,</p><p>)</p><p> </p><p>your call.</p><p> </p><p>  </p>

Reply:
Oh, that's cool. I didn't realize you could use imports that way. Learned something new :)
I'll give it a shot and see how it looks.